File: src/SatelliteTracker.Database/Entities/Pass.cs
namespace SatelliteTracker.Database.Entities;

public class Pass
{
    public Guid Id { get; set; }
    public Guid SatelliteId { get; set; }
    public Guid TleId { get; set; }
    public int OrbitNumber { get; set; }
    public DateTime Aos { get; set; }
    public DateTime Los { get; set; }
    public decimal MaxElevation { get; set; }
    public decimal AosAzimuth { get; set; }
    public decimal LosAzimuth { get; set; }
    public int DurationSec { get; set; }
    public bool NotificationSent { get; set; }
    public DateTime? NotificationSentAt { get; set; }
    public bool OutlookSynced { get; set; }
    public DateTime CalculatedAt { get; set; }

    public Satellite Satellite { get; set; } = null!;
    public TleRecord TleRecord { get; set; } = null!;
    public ICollection<Note> Notes { get; set; } = [];
}

File: android/app/src/main/java/com/sattrakk/app/domain/model/Pass.kt
package com.sattrakk.app.domain.model

import java.time.OffsetDateTime

// `notify` has no equivalent on the backend's PassDto — see PassEntity's doc comment for why it's
// carried here as local-only, tester-toggleable state.
data class Pass(
    val id: String,
    val satelliteId: String,
    val tleId: String,
    val orbitNumber: Int,
    val aos: OffsetDateTime,
    val los: OffsetDateTime,
    val maxElevation: Double,
    val aosAzimuth: Double,
    val losAzimuth: Double,
    val durationSec: Int,
    val notify: Boolean,
    val outlookSynced: Boolean,
    val calculatedAt: OffsetDateTime
)
